internal/app/services.go

```diff
@@ -109,7 +109,7 @@ func InitializeServices(cfg *Config) (*Services, error) {
 
 	// Step 1: Create unified muster client once
 	// This avoids redundant Kubernetes connection attempts and CRD validation
-	musterClient, err := createMusterClient(cfg.ConfigPath, cfg.Debug)
+	musterClient, err := createMusterClientWithConfig(cfg.ConfigPath, cfg.Debug, *cfg.MusterConfig)
```

Comment:
This is a typical problem with the agent. Please make sure there is only one way to create the client and not multiple.